infra/src/test/kotlin/fi/fta/geoviite/infra/ui/testgroup1/UserRoleTestUI.kt

```diff
@@ -68,30 +104,80 @@ fun assertPublicationLog(frontpage: E2EFrontPage) {
         .returnToFrontPage()
 }
 
-fun assertMapPage(role: E2ERole) {
+fun assertMapPage(role: E2ERole, trackNumber: TrackNumber, designName: String) {
     val mapPage = E2EAppBar().goToMap()
 
     when (role) {
         E2ERole.Operator,
         -> {
             mapPage
+                .also { assertDraftAndDesignModeTabsVisible() }
```

Comment:
Minulle tämä mapPageen liittymättömien asioiden also-ketjutus näissä näyttää varsin oudolta eikä yhtään luettavammalta kuin ihan vain kutsua noita (ja toistaa sitten se mapPage kun sitä taas tarvitaan). Tämä assert ei selvästikään käytä mapPagea yhtään mitenkään, joten pysäytti lukemaan parikin kertaa että miksi tämä on näin. 

Tuolla alempana esiim tuo toolbarista tehtävä valinta sentään käyttää mapPagea johonkin, mutta also-ketjussa se renameutuu `mapPage` -> `it`, mikä ei sekään ole mielestäni yhtään selvempi vaan päinvastoin pakottaa kurkkaamaan monta riviä ylempää että millekäs oliolle tätä kutsutaankaan. Toki tuollainen ketjutus on näissä testeissä konventio, mikä selkeyttää asiaa ja säästäähän se muutaman merkin kun mapPagea ei tarvitse toistaa noille muille operaatioille. 

Yleisesti pidän tuota ketjutusta kyllä vähän so-so käytäntönä. Onhan siinä vähemmän kohinaa kun tuota olion nimeä ei toistella, mikä helpottaa lukemista. Toisaalta jos välissä olisi yksikin funktio joka palauttaa jotain muuta kuin alkuperäisen olion itsensä, sitä ei näkisi tuossa ketjussa mistään ja tuon ymmärtäminen menisi heti  ^2 vaikeammaksi. 

Ei siis kannata lähteä itse konventiota tässä muuttamaan, tämä lähinnä mutuilu-huomiona. Mutta en oikein pidä siitä että täysin mapPageen liittymättömätkin asiat ketjutetaan also:lla samaan ihan vain just-because.